### PR TITLE
Align the `-server.http-write-timeout` with `-querier.timeout` to 2m

### PR DESCRIPTION
**Pull request by hi-rustin**
#### What this PR does
Align the `-server.http-write-timeout` with `-querier.timeout` to 2m. Add validation for these values.
#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/mimir/issues/3248

#### Checklist

- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/querier/querier.go
Comment (dimitarvdimitrov) [inline pkg/querier/querier.go:91]: What do you think about moving this into the `pkg/mimir` package? Along with the [other validations](https://github.com/grafana/mimir/blob/cca6e117cc537e4b68f5d31e27ca7c7eeda43618/pkg/mimir/mimir.go#L207-L212) that rely on multiple configuration structs?

I think that way we'll have easier testing and arguably maybe better separation of concerns in the querier package.

**Review by dimitarvdimitrov — COMMENTED** [review]
Thanks for picking this up :) I think we can move it into a higher level package

File: pkg/querier/querier.go
Comment (hi-rustin) [reply pkg/querier/querier.go]: Godd sugesstion! Thanks!

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM <3

I'll let others have a look before merging

File: CHANGELOG.md
Comment (pracucci) [inline CHANGELOG.md:23]: We use CHANGE for any change to default config. Please move it above among other CHANGE entries.

```suggestion
* [CHANGE] The default value of `-server.http-write-timeout` has changed from 30s to 2m. #3346
```

File: pkg/mimir/mimir.go
Comment (pracucci) [inline pkg/mimir/mimir.go:229]:
```suggestion
		return fmt.Errorf("querier timeout (%s) must be lower than or equal to HTTP server write timeout (%s)",
```

**Review by pracucci — APPROVED** [review]
LGTM, thanks! I left a couple of nits I'm going to auto apply.

File: pkg/mimir/mimir.go
Comment (hi-rustin) [reply pkg/mimir/mimir.go]: Good catch! 
Thanks for your review!  💚 💙 💜 💛 ❤️

**Pull request merged**
